pages/radio-group/progressive-disclosure.page.tsx

```diff
@@ -0,0 +1,113 @@
+import Select, { SelectProps } from '~components/select';
+
+const langOptions: Array<SelectProps.Option> = [
+  { value: 'en-US', label: 'English (United States)' },
+  { value: 'en-GB', label: 'English (United Kingdom)' },
+  { value: 'en-NZ', label: 'English (New Zealand)' },
+  { value: 'de-DE', label: 'Deutsch (Deutschland)' },
+  { value: 'de-CH', label: 'Deutsch (Schweiz)' },
+  { value: 'de-AT', label: 'Deutsch (Österreich)' },
+];
+
+const optionsPerLanguage = langOptions.reduce((acc, option) => {
+  const [lang] = option.value!.split('-');
+  if (!acc[lang]) {
+    acc[lang] = [];
+  }
+  acc[lang].push(option);
+  return acc;
+}, {} as Record<string, Array<SelectProps.Option>>);
+
+const initialState = {
+  en: optionsPerLanguage.en[0],
+  de: optionsPerLanguage.de[0],
+};
+
+type State = { [K in keyof typeof initialState]: SelectProps.Option };
+interface Action {
+  type: `update/${string & keyof State}`;
+  payload: Partial<State[keyof State]>;
+}
+
+const langReducer: React.Reducer<State, Action> = (state = initialState, action) => {
+  switch (action.type) {
+    case 'update/en':
+      return { ...state, en: action.payload };
+    case 'update/de':
+      return { ...state, de: action.payload };
+    default:
+      return state;
+  }
+};
+
+export default function RadiosPage() {
+  const [radioSelection, setRadioSelection] = useState<string>('');
+
+  const [lang, dispatch] = useReducer(langReducer, initialState);
+  const setLang = (type: keyof State, option: SelectProps.Option) =>
+    dispatch({ type: `update/${type}`, payload: option });
+
+  return (
+    <Box padding="l">
+      <h1>Radio group progressive disclosure demo</h1>
+      <FormField
+        label="Language settings"
+        description="You can select a specific language or have this service automatically identify the language in your media."
+      >
+        <RadioGroup
+          value={radioSelection}
+          onChange={event => setRadioSelection(event.detail.value)}
+          ariaControls="dialekt-settings"
+          items={[
+            {
+              label: 'English',
+              value: 'en',
+              description:
+                'If you know the language spoken in the source media is English, choose this option for optimal results.',
+            },
+            {
+              label: 'Deutsch',
+              value: 'de',
+              description:
+                'Wenn Sie wissen, dass die in den Quellmedien gesprochene Sprache Deutsch ist, wählen Sie diese Option für optimale Ergebnisse.',
+            },
+            {
+              label: 'Automatic language detection',
+              value: 'automatic',
+              description: 'If you do not know the language spoken in the source media, choose this option.',
+            },
+          ]}
+        />
+      </FormField>
+      <Box id="dialekt-settings" display={!Object.keys(lang).includes(radioSelection) ? 'none' : 'block'}>
```

Comment:
Should it be `display:none`, `aria-hidden=true` or just a conditional rendering? What happens if aria controls resolves to a non-existing id?

Reply:
Can we do conditional rendering? (I think this should be recommended, for example to allow lazy-loading options for conditional select). If not, what is the alternative?

Reply:
> What happens if aria controls resolves to a non-existing id?

Certain a11y checkers will throw a violation 👀

Reply:
Exactly, the referenced ID is not in the document; some checkers will throw a violation. <sup>[example](https://rocketvalidator.com/html-validation/the-aria-controls-attribute-must-point-to-an-element-in-the-same-document)</sup>

Reply:
How is it different if hidden elements are not in the a11y tree anyway?

Reply:
P.S. if you have an expert approval on this pattern, I can approve the PR after addressing other comments.

But my general software development experience suggests me that something is does not match up here

Reply:
`aria-controls` is not limited to visible elements; at least as far as the spec is concerned.

https://www.w3.org/TR/wai-aria-1.1/#aria-controls
For example, Tab content visibility is given as a valid use of aria-controls

In the discussion for this issue, they make the distinction between non-existent and non-visible elements.
https://github.com/w3c/aria/issues/531